crates/re_viewer_context/src/space_view/space_view_class_impl.rs

```diff
@@ -1,21 +1,27 @@
-use crate::{Scene, SpaceViewClass, SpaceViewClassName, SpaceViewState, ViewerContext};
+use crate::{
+    Scene, SceneContext, ScenePartCollection, SpaceViewClass, SpaceViewClassName, SpaceViewState,
+    ViewerContext,
+};
 
-use super::scene_element_list::SceneElementListConversionError;
+use super::scene::TypedScene;
 
-/// Utility for implementing [`SpaceViewClass`] with concrete [`SpaceViewState`] and [`crate::SceneElement`] type.
+/// Utility for implementing [`SpaceViewClass`] with concrete [`SpaceViewState`] and [`crate::ScenePart`] type.
 ///
 /// Each Space View in the viewer's viewport has a single class assigned immutable at its creation time.
 /// The class defines all aspects of its behavior.
 /// It determines which entities are queried, how they are rendered, and how the user can interact with them.
-pub trait SpaceViewClassImpl {
+pub trait SpaceViewClassImpl: std::marker::Sized {
     /// State of a space view.
-    type State: SpaceViewState + Default + 'static;
+    type SpaceViewState: SpaceViewState + Default + 'static;
 
-    /// A tuple of [`crate::SceneElement`] types that are supported by this space view class.
-    type SceneElementTuple: Into<Scene>
-        + TryFrom<Scene, Error = SceneElementListConversionError>
-        + Default
-        + 'static;
+    /// Context of the scene, which is passed to all [`crate::ScenePart`]s and ui drawing on population.
+    type SceneContext: SceneContext + Default + 'static;
+
+    /// Collection of [`crate::ScenePart`]s that this scene populates.
+    type ScenePartCollection: ScenePartCollection<Self> + Default + 'static;
+
+    /// A piece of data that all scene parts have in common, useful for iterating over them.
+    type ScenePartData;
```

Comment:
It's unclear to me how this differs from `SpaceViewState`. As I understand it, `SpaceViewState` is also data that all scene parts have in common